pkgs/applications/window-managers/wayfire/wf-shell.nix

```diff
@@ -0,0 +1,72 @@
+{ stdenv
+, fetchFromGitHub
+, fetchFromGitLab
+, meson
+, ninja
+, pkgconfig
+, wayland
+, wayland-protocols
+, wf-config
+, gtkmm3
+, gobject-introspection
+, libpulseaudio
+, alsaLib
+, libjpeg
+, gtk-layer-shell
+}:
+
+let
+  # fetch submodule
+  libgnome-volume-control = fetchFromGitLab {
+    rev = "468022b708fc1a56154f3b0cc5af3b938fb3e9fb";
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "libgnome-volume-control";
+    sha256 = "01wrl875r52v7f12yrx3qdn2d2s3h2xz7jx1dswydypd9b6qss0d";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "wf-shell";
+  version = "0.3";
+
+  src = fetchFromGitHub {
+    owner = "WayfireWM";
+    repo = pname;
+    rev = version;
+    sha256 = "0xgr5lgrhdab3qm24z0ws1b7xz1mhf6m1gxzgj5wpr4pc7iwc7lw";
```

Comment:
Use `fetchSubmodules = true` here rather than doing it by hand.

Reply:
For some reason, `fetchSubmodules = true` doesn't seem to work. It fails with:
```
meson.build:22:0: ERROR: Subproject exists but has no meson.build file
```
Running with `--keep-failed` and looking in `/tmp/nix-build-wf-shell-0.3.drv-0/source/subprojects/`, both subdirectories are empty. I wasn't originally aware of `fetchSubmodules`: I looked in nixpkgs for an example to copy, and that's how I ended up doing it the way I did. Not sure if the example I looked at had a good reason to do it by hand or should have been using `fetchSubmodules`.

I was originally wondering whether it be better to de-vendor the remaining submodule, 'gvc' (The other submodule, gtk-layer-shell is already a separate buildInput, but I suspect it would be ignored if the submodule were also present).